### PR TITLE
[Storage] Tests - common naming - shares.

### PR DESCRIPTION
**Pull request by kasobol-msft**

### CODE REVIEW

File: sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileServiceAPITests.groovy [hunk 1]
```diff
@@ -276,7 +276,6 @@ class FileServiceAPITests extends APISpec {
         FileTestHelper.assertFileServicePropertiesAreEqual(updatedProperties, getPropertiesAfterResponse.getValue())
     }
 
-    @Requires( { playbackMode() } )
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileServiceAPITests.groovy:279]: I find this disturbing.

File: sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/APISpec.groovy [hunk 2]
```diff
@@ -114,33 +89,34 @@ class APISpec extends StorageSpec {
...
-            for (def share : cleanupFileServiceClient.listShares(new ListSharesOptions().setPrefix(methodName.toLowerCase()), null, Context.NONE)) {
-                def shareClient = cleanupFileServiceClient.getShareClient(share.getName())
-
-                if (share.getProperties().getLeaseState() == LeaseStateType.LEASED) {
-                    createLeaseClient(shareClient).breakLeaseWithResponse(new ShareBreakLeaseOptions().setBreakPeriod(Duration.ofSeconds(0)), null, null)
-                }
-
-                shareClient.deleteWithResponse(new ShareDeleteOptions().setDeleteSnapshotsOptions(ShareSnapshotsDeleteOptionType.INCLUDE), null, null)
+        def cleanupFileServiceClient = new ShareServiceClientBuilder()
+            .connectionString(connectionString)
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/APISpec.groovy:99]: should we add the retry options?

Comment (kasobol-msft) [reply sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/APISpec.groovy:99]: there's exponential retry by default. what was here before was causing 5xx timeouts...

**Review by gapra-msft — APPROVED** [review]

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**